employees/tests/test_unit_custom_mixins.py

```diff
@@ -14,33 +18,64 @@
 
 class MonthNavigationMixinCustomMethodsTests(TestCase):
     def setUp(self):
+        self.factory = RequestFactory()
         self.mixin = MonthNavigationMixin()
+        self.current_date = datetime.datetime.now().date()
 
     def test_get_url_from_date_should_return_url_with_no_pk_if_none_is_provided(self):
-        current_date = datetime.datetime.now().date()
-        self.mixin.url_name = "custom-report-list"
+        request = self.factory.get(
+            reverse("custom-report-list", kwargs={"year": self.current_date.year, "month": self.current_date.month})
+        )
+        self.mixin.request = request
         self.assertEqual(
-            self.mixin._get_url_from_date(current_date, None), f"/reports/{current_date.year}/{current_date.month}/"
+            self.mixin._get_url_from_date(self.current_date, None),
+            f"/reports/{self.current_date.year}/{self.current_date.month}/",
         )
 
     def test_get_title_date_should_return_five_character_string_containing_month_number_separator_and_two_last_digits_of_year_number(
         self
     ):
-        self.assertEqual(self.mixin._get_title_date(year=2019, month=4), "04/19")
+        self.assertEqual(self.mixin._get_title_date(year=2019, month=12), f"12/19")
 
     def test_get_next_month_url_should_generate_url_for_next_month_for_given_path(self):
-        self.mixin.url_name = "project-report-list"
-        self.assertEqual(self.mixin._get_next_month_url(2018, 12, 1), "/reports/project/1/2019/1/")
+        request = self.factory.get(
+            reverse(
+                "project-report-list",
+                kwargs={"year": self.current_date.year, "month": self.current_date.month, "pk": 1},
+            )
+        )
+        self.mixin.request = request
+        next_month = self.current_date + relativedelta(months=+1)
+        self.assertEqual(
+            self.mixin._get_next_month_url(self.current_date.year, self.current_date.month, 1),
+            f"/reports/project/1/{next_month.year}/{next_month.month}/",
```

Comment:
Same here, also in few places below, please update them.

Reply:
done